### PR TITLE
Rename AS observe to query

### LINKED ISSUE #647: Change AS verb "observe" to "query"
> I'm proposing to change it to "query" instead, because the term 1) makes more sense in the language itself and 2) is better suited for many more query types.

Background is adding query types for room info (#540) and other things that are not room attendance. And since the next release will break AS object notations, this would be a good time to introduce more breaking changes all at once to have avoid more breakage in the future.

### PR DESCRIPTION
**Pull request by silverbucket**
This is ready for review but is based on #671 which should be reviewed & merged first.

Resolves #647

### CODE REVIEW

File: packages/sockethub-platform-irc/index.js [hunk 1]
```diff
@@ -417,9 +417,9 @@ IRC.prototype.update = function (job, credentials, done) {
 };
 
 /**
- * Function: observe
+ * Function: query
  *
- * Indicate an intent to observe something (ie. get a list of users in a room).
+ * Indicate an intent to query something (ie. get a list of users in a room).
```
Comment (raucao) [inline packages/sockethub-platform-irc/index.js]:
```suggestion
 * Indicate an intent to query something (e.g. get a list of users in a room).
```

**Comment by raucao** [comment]
Using the old term "observe" results in the following Sockethub error message being sent to the client, instead of a validation error:

> TypeError: platform[jobData.msg.type] is not a function"

**Comment by raucao** [comment]
One more thing I found: the XMPP platform is still sending the attendance list as type "observe" instead of "query".

**Comment by silverbucket** [comment]
@raucao OK, this should be ready for review again. There weren't any up-front checks for platform method validation before being called, so this ended up touching a lot more files to get that in place. Fixed some tests too. Should be a lot more stable now.

**Review by raucao — APPROVED** [review]
Works great now!

**Pull request merged**